### PR TITLE
Backport pg_stat_progress_analyze|basebackup|copy

### PR DESCRIPTION
**Pull request by huansong**
Similar to https://github.com/greenplum-db/gpdb/pull/14960, this is just per-segment view. Also currently they should work for heap table only. Example output (GPDB is in utility mode, connecting to one of the primaries):

```
--analyze:
-- PG15:
postgres=# select * from pg_stat_progress_analyze;
 pid  | datid | datname  | relid |         phase         | sample_blks_total | sample_blks_scanned | ext_stats_total | ext_stats_computed | child_tables_total | child_tables_done | current_child_table_relid
------+-------+----------+-------+-----------------------+-------------------+---------------------+-----------------+--------------------+--------------------+-------------------+---------------------------
 9145 | 12796 | postgres | 16438 | acquiring sample rows |             30000 |               20882 |               0 |                  0 |                  0 |                 0 |                         0
(1 row)

-- GPDB7:
postgres=# select * from pg_stat_progress_analyze;
  pid  | datid | datname  | relid |         phase         | sample_blks_total | sample_blks_scanned | ext_stats_total | ext_stats_computed | child_tables_total | child_tables_done | current_child_table_relid
-------+-------+----------+-------+-----------------------+-------------------+---------------------+-----------------+--------------------+--------------------+-------------------+---------------------------
 21480 | 13379 | postgres | 16384 | acquiring sample rows |             30000 |                2121 |               0 |                  0 |                  0 |                 0 |                         0
(1 row)


--basebackup:
-- PG15:
postgres=# select * from pg_stat_progress_basebackup;
  pid  |          phase           | backup_total | backup_streamed | tablespaces_total | tablespaces_streamed
-------+--------------------------+--------------+-----------------+-------------------+----------------------
 22976 | streaming database files |   4034732032 |      2164750336 |                 1 |                    0
(1 row)

-- GPDB7:
postgres=# select * from pg_stat_progress_basebackup;
  pid  |          phase           | backup_total | backup_streamed | tablespaces_total | tablespaces_streamed
-------+--------------------------+--------------+-----------------+-------------------+----------------------
 31775 | streaming database files |    423681536 |       255780352 |                 1 |                    0
(1 row)



--copy:
-- PG15:
postgres=# select * from pg_stat_progress_copy;
 pid  | datid | datname  | relid | command | type | bytes_processed | bytes_total | tuples_processed | tuples_excluded
------+-------+----------+-------+---------+------+-----------------+-------------+------------------+-----------------
 3866 | 12796 | postgres | 16396 | COPY TO | FILE |        41362493 |           0 |          5339476 |               0
(1 row)

-- GPDB7:
postgres=# select * from pg_stat_progress_copy;
  pid  | datid | datname  | relid | command | type | bytes_processed | bytes_total | tuples_processed | tuples_excluded
-------+-------+----------+-------+---------+------+-----------------+-------------+------------------+-----------------
 13218 | 13392 | postgres | 16384 | COPY TO | FILE |        25040728 |           0 |          3268979 |               0
(1 row)
```

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/include/commands/copy.h [hunk 1]
```diff
@@ -175,6 +175,7 @@ typedef struct CopyStateData
 	 */
 	FmgrInfo   *out_functions;	/* lookup info for output functions */
 	MemoryContext rowcontext;	/* per-row evaluation context */
+	uint64	bytes_processed;	/* number of bytes processed so far */
```
Comment (soumyadeep2007) [inline src/include/commands/copy.h:178]: Let's leave an instruction for 14 merge with a `GPDB_14_MERGE_FIXME: Move this to CopyToStateData and CopyFromStateData`

**Review by soumyadeep2007 — COMMENTED** [review]
Looks good. I have 2 minor items.

I think we missed this [hunk](https://github.com/postgres/postgres/commit/8a4f618e7ae3cb11b0b37d0f06f05c8ff905833f#diff-5974661d4ffb034fd936898b834cfd5c28e5ada5ea84b8863a32355af35c4f5bR1536) in `EndCopyFrom`.

**Review by soumyadeep2007 — APPROVED** [review]

**Comment by soumyadeep2007** [comment]
Also we may need a catversion bump. No?

**Comment by huansong** [comment]
> I think we missed this [hunk](https://github.com/postgres/postgres/commit/8a4f618e7ae3cb11b0b37d0f06f05c8ff905833f#diff-5974661d4ffb034fd936898b834cfd5c28e5ada5ea84b8863a32355af35c4f5bR1536) in EndCopyFrom.

I didn't take this because `EndCopy` would have done the same in current GPDB7 code. But we probably can just add it in `EndCopyFrom` too because that'll help avoid missing it when we make `EndCopyFrom` same as upstream in future. Doing it twice should be OK as the second time should be a no-op. Followed your another suggestion and added a `GPDB_14_MERGE_FIXME` here.

> Let's leave an instruction for 14 merge with a GPDB_14_MERGE_FIXME: Move this to CopyToStateData and CopyFromStateData

Done!

> Also we may need a catversion bump. No?

Yes, did that.

**Review by l-wang — APPROVED** [review]
LGTM upon resolving catversion conflict.

**Comment by l-wang** [comment]
As I was reviewing this PR I was surprised upstream did not add a single test for these views. I guess it's indeed difficult to write an upstream test case w/o the fault injection framework, and since this is only backport I'm fine with what we have here. I'm very happy with tests Deep added for AO/CO tables in https://github.com/greenplum-db/gpdb/pull/14680

**Comment by huansong** [comment]
> As I was reviewing this PR I was surprised upstream did not add a single test for these views. I guess it's indeed difficult to write an upstream test case w/o the fault injection framework, and since this is only backport I'm fine with what we have here. I'm very happy with tests Deep added for AO/CO tables in https://github.com/greenplum-db/gpdb/pull/14680

I was thinking about adding tests when creating the cluster-wise views of these (mentioned that in https://github.com/greenplum-db/gpdb/pull/14960 but not the current PR). But yeah definitely possible to do so with fault injection framework we have.

**Pull request merged**